docs/ref/models/database-functions.txt

```diff
@@ -293,3 +293,518 @@ Usage example::
 .. versionchanged:: 1.9
 
     The ability to register the function as a transform was added.
+
+Date Functions
+==============
+
+.. module:: django.db.models.functions.date
+
+.. versionadded:: 1.10
+
+We'll be using the following model in examples of each function::
+
+    class Experiment(models.Model):
+        start_time = models.DateTimeField()
+        end_time = models.DateTimeField(null=True, blank=True)
+
+``Extract``
+-----------
+
+.. class:: Extract(date_expression, lookup_name=None, **extra)
```

Comment:
I intentionally renamed it to make the docs clearer that the type of expression should resolve to a date. However, it'd be more accurate to rename to `datetime_expression` so that it fits with functions that only accept a datetime, like Hour/Minute/Second. 

Alternatively, I can rename to `expression` since the following paragraph clarifies the type that expression needs to resolve to. I think that might actually be nicer to read.